sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BulkExecutor.java

```diff
@@ -537,15 +538,30 @@ private int calculateTotalSerializedLength(AtomicInteger currentTotalSerializedL
                                 getThreadInfo());
                             return this.enqueueForRetry(result.backOffTime, groupSink, itemOperation, thresholds);
                         } else {
-                            logger.error(
-                                "HandleTransactionalBatchOperationResult - Fail, PKRange {}, Response Status " +
-                                    "Code {}, Operation Status Code {}, {}, Context: {} {}",
-                                thresholds.getPartitionKeyRangeId(),
-                                response.getStatusCode(),
-                                operationResult.getStatusCode(),
-                                getItemOperationDiagnostics(itemOperation),
-                                this.operationContextText,
-                                getThreadInfo());
+                            // reduce log noise level for commonly expected/normal status codes
+                            if (response.getStatusCode() == HttpConstants.StatusCodes.CONFLICT ||
+                                response.getStatusCode() == HttpConstants.StatusCodes.PRECONDITION_FAILED) {
+
+                                logger.debug(
```

Comment:
nit: This is only noisy for Spark, but not noisy for a common Java Bulk scenario though IMO

Reply:
Customers would still need to check status code - whether we log it or not - so the right place to decide whether it is noisy or not is at that level. I made this change because every single use case doing ingestion at bulk will need to make an explicit decision how to handle them (throw, ignore, retry) - so not worth logging at error level for each item IMO.